pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -1194,6 +1222,21 @@ func storageTypePtr(storageType string) *compute.StorageAccountTypes {
 	return &storage
 }
 
+func ipSkuPtr(ipSKU string) *network.PublicIPAddressSkuName {
```

Comment:
Not a blocker. 

nit: This looks too complicated for what it does. 

https://pkg.go.dev/strings#Title

It is deprecated but probably okay for us. Or use golang.org/x/text/cases suggested there instead. 

--- 

Maybe we could just refuse improperly typed sku during validation.

Reply:
I didn't want to use the deprecated method because, well, it's deprecated, and I didn't want to introduce an additional package either.

IMHO we shouldn't reject the SKUs without title case because we allow it in KKP as well.